tool/simulation/change_parameters.py

```diff
@@ -0,0 +1,255 @@
+        return iter_files( kwargs["rest_files"], **copy_kwargs )
+
+    replace_key = next(iter(copy_paras)) # Take the first key to replace
+    vals = copy_paras.pop(replace_key)
+    for val in vals:
+        if not kwargs["quite"]: print("File <%s> changing %s --> %s"%(file_name, replace_key, str(val)))
+        replace_parameter( file_name, replace_key, val, flag )
+        record[replace_key]=val
+        iter_paras( file_name, copy_paras, record, flag, **kwargs )  # repalce the next parameter
+    return
+
+
+def replace_parameter( file_name, name, val, flag ):
+    if flag:
+        with open( file_name, 'r' ) as f:
+            header = f.readline()
+        index = { v:i for i, v in enumerate(header.split()[1:]) }       # The first element assume to be "#"
+        data = np.loadtxt(file_name)
+        if len(index) != data.shape[1]: raise BaseException("ERROR: The number of columns in <%s> does not match to the header."%(file_name))
+        data[:, index[name]] = val
+        np.savetxt(file_name, data, fmt="%7g"+"%20.16g"*(len(index)-1), header=header[2:-1])
+    else:
+        with open( file_name, 'r' ) as f:
+            content = f.read()
+
+        content_new, n_sub = re.subn( r"^(%s\s+)([^\s]+)"%name, r"\g<1>%s"%str(val), content, flags=re.M )
+        if n_sub == 0: raise BaseException("ERROR: Can not find <%s> in <%s>."%(name, file_name))
+
+        with open( file_name, 'w' ) as f:
+            f.write( content_new )
+    return
+
+def execution( **kwargs ):
+    record_parameters = kwargs["record"]
+    print("GO GO GAMER GO!")
+    return RETURN_SUCCESS
+
+
+
+#====================================================================================================
+# Main
+#====================================================================================================
+sys.setrecursionlimit(RECURSION_LIMIT)                                          # Reset the recursion depth limit
+
+# 1. Set up the files and the parameters to be changed
+file_name1   = "Input__Parameter"                                               # file name
+const_paras1 = {"END_T":-1, "END_STEP":50}                                      # the parameters change only once
+loop_paras1  = {"NX0_TOT_X":[16, 32], "NX0_TOT_Y":[16, 32]}                     # the parameters change as given list
+# set the `File` class, the file changing method is decided by the last parameter. If the file is a flag file, please set it to `True`.
+file1        = File( file_name1, const_paras1, loop_paras1, flag=False )
+
+file_name2   = "Input__Testprob"
+const_paras2 = {"Const_1":-1, "Const_2":50}
+loop_paras2  = {"Parameter_1":[0.01, 0.02], "Parameter_2":[16, 32]}
+file2        = File( file_name2, const_paras2, loop_paras2, flag=False )
+
+file_name3   = "Input__Flag"
+const_paras3 = {"derefine":0.1}
+loop_paras3  = {"Soften":[0.01, 0.02], "refine":[16, 32]}
+file3        = File( file_name3, const_paras3, loop_paras3, flag=True )
+
```

Comment:
Here maybe can add an example demonstrating how to modify a column, showing how to change individual values within a column rather than assigning the same value to all entries.
Like this:
```
file_name4       = "Input__Flag_Rho"
const_paras4     = {}
loop_paras4      = { "Density": [(2,4,8,16,32,64,128,256,512,1024),(4,8,16,32,64,128,256,512,1024,2048)]}
file4            = File( file_name4, const_paras4, loop_paras4, flag=True )
```

Reply:
Good point! I have updated it accordingly.